otherlibs/unix/fork.c

```diff
@@ -19,14 +19,21 @@
 #include <caml/debugger.h>
 #include <caml/eventlog.h>
 #include "unixsupport.h"
+#include <caml/domain.h>
+#include <caml/fail.h>
 
 CAMLprim value unix_fork(value unit)
 {
   int ret;
+  if (caml_domain_is_multicore()) {
+    caml_failwith
```

Comment:
Should it be `failwith` or `invalid_argument` or a Unix error code like `ENOTSUPP` ?  I don't know yet, just making a note to think about it later.

Reply:
how about spawn.c when using the fallback implementation based on fork? Is that safe with multicore, or should it similarly raise an exception? (I assume the posix_spawn based implementation would be safe)

Reply:
See my previous reply here: https://github.com/ocaml-multicore/ocaml-multicore/issues/801#issuecomment-996816236